service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapper.java

```diff
@@ -0,0 +1,155 @@
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.github.mustachejava.Mustache;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ConditionLinkSetMapper {
+
+    private static final Mustache OBSERVATION_STATEMENT_TEMPLATE = TemplateUtils
+        .loadTemplate("ehr_link_set_template.mustache");
+    private static final String ACTUAL_PROBLEM_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-ActualProblem-1";
+    private static final String PROBLEM_SIGNIFICANCE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-ProblemSignificance-1";
+    private static final String RELATED_CLINICAL_CONTENT_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-RelatedClinicalContent-1";
+    private static final String ACTIVE = "active";
+    private static final String MAJOR = "major";
+    private static final String LIST = "List";
+    private static final String EXTENSION_NOT_PRESENT = "Extension not present";
+
+    private final MessageContext messageContext;
+
+    public String mapConditionToLinkSet(Condition condition, boolean isNested) {
+        var builder = ConditionLinkSetMapperParameters.builder()
+            .isNested(isNested)
+            .linkSetId(messageContext.getIdMapper().getOrNew(ResourceType.Condition, condition.getIdElement().getIdPart()));
+
+        buildEffectiveTimeLow(condition).map(builder::effectiveTimeLow);
+        buildEffectiveTimeHigh(condition).map(builder::effectiveTimeHigh);
+        buildAvailabilityTime(condition).map(builder::availabilityTime);
+        builder.relatedClinicalContent(buildRelatedClinicalContent(condition));
+
+        var qualifier = buildQualifier(condition);
+        qualifier.map(value -> value.equalsIgnoreCase(MAJOR)).map(builder::qualifierIsMajor);
+        qualifier.map(builder::qualifier);
+
+        var clinicalStatus = buildClinicalStatusCode(condition);
+        clinicalStatus.map(value -> value.equalsIgnoreCase(ACTIVE)).map(builder::clinicalStatusIsActive);
+        clinicalStatus.map(builder::clinicalStatusCode);
+
+        var conditionNamed = buildConditionNamed(condition);
+        if (conditionNamed.isPresent()) {
+            conditionNamed
+                .filter(value -> !value.equalsIgnoreCase(EXTENSION_NOT_PRESENT))
+                .map(builder::conditionNamed);
+        } else {
+            String newId = messageContext.getIdMapper().getNew();
+            builder.generateObservationStatement(true);
+            builder.conditionNamed(newId);
+            buildPertinentInfo(condition).map(builder::pertinentInfo);
+        }
```

Comment:
Using above suggestion above and ifPresentOrElse this could be simplified:

```suggestion
buildConditionNamed(condition)
            .ifPresentOrElse(builder::conditionNamed, () -> {
                String newId = messageContext.getIdMapper().getNew();
                builder.generateObservationStatement(true);
                builder.conditionNamed(newId);
                buildPertinentInfo(condition).ifPresent(builder::pertinentInfo);
            });
```

Not sure why the test fails - there is either an extension referencing an observation or there is not.